spec/integration/thread_safety_spec.rb

```diff
@@ -0,0 +1,15 @@
+require 'spec_helper'
+
+describe 'handling many concurrent schema changes' do
+  it 'does not fail first when many jobs are writing concurrently' do
+    @error = false
+    use_real_backend { |config| config.error_notifier = proc { @error = true } }
+
+    LockerHelper.delay_after_unlock(0.5) do
```

Comment:
easiest to test this case by delaying immediately after releasing a lock. any safe code should be fine, but any code that should be locked but isn't is hosed